packages/react/src/components/AmplifyProvider/AmplifyContext.tsx

```diff
@@ -1,6 +1,19 @@
 import { createContext } from 'react';
 
-export const AmplifyContext = createContext({
+import { defaultTheme, Theme } from '../../theming';
+import { getCSSVariablesFromTheme } from '../../theming';
+
+interface AmplifyContextType {
+  components: any;
+  theme: {
+    themeObject: unknown;
```

Comment:
I made a draft PR to generate a Theme type from style dictionary where it marks all parts of the default theme as optional. We could use that here? https://github.com/aws-amplify/amplify-ui/pull/312

Reply:
That's a good question which I have brought up earlier above. The overall implementation so far and the generic types you saw is for providing users with the ability to introduce their own theme in case some of them don't want our styling. Any thoughts?